src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```diff
@@ -602,77 +605,168 @@ private static String EscapeIDName(String inID) throws SQLServerException {
+            pstmt.closeOnCompletion();
+            try {
+                pstmt.setString(1, (null != table && !table.isEmpty()) ? table : "%");
+                pstmt.setString(2, (null != schema && !schema.isEmpty()) ? schema : "%");
+                pstmt.setString(3, (null != catalog && !catalog.isEmpty()) ? catalog : this.connection.getCatalog());
+                pstmt.setString(4, (null != col && !col.isEmpty()) ? col : "%");
+                pstmt.setInt(5, 2);// show sparse columns
+                pstmt.setInt(6, 3);// odbc version
+
+                rs = (SQLServerResultSet) pstmt.executeQuery();
+                rs.getColumn(5).setFilter(new DataTypeFilter());
+                rs.getColumn(7).setFilter(new ZeroFixupFilter());
+                rs.getColumn(8).setFilter(new ZeroFixupFilter());
+                rs.getColumn(16).setFilter(new ZeroFixupFilter());
+            } catch (SQLException e) {
+                errorOnClose = e;
+                pstmt.close();
+            } finally {
+                if (null != originalCatalog) {
+                    connection.setCatalog(originalCatalog);
+                }
+                if (null != errorOnClose) {
+                    throw errorOnClose;
+                }
+            }
 
-        // sp_columns supports wild carding schema table and columns
-        String column = EscapeIDName(col);
-        table = EscapeIDName(table);
-        schema = EscapeIDName(schema);
-
-        /*
-         * sp_columns [ @table_name = ] object [ , [ @table_owner = ] owner ] [ , [ @table_qualifier = ] qualifier ] [ ,
-         * [ @column_name = ] column ] [ , [ @ODBCVer = ] ODBCVer ]
-         */
-        String[] arguments;
-        if (connection.isKatmaiOrLater())
-            arguments = new String[6];
-        else
-            arguments = new String[5];
-        arguments[0] = table;
-        arguments[1] = schema;
-        arguments[2] = catalog;
-        arguments[3] = column;
-        if (connection.isKatmaiOrLater()) {
-            arguments[4] = "2"; // give information about everything including
-                                // sparse columns
-            arguments[5] = "3"; // odbc version
-        } else
-            arguments[4] = "3"; // odbc version
-        SQLServerResultSet rs;
-        if (connection.isKatmaiOrLater())
-            rs = getResultSetWithProvidedColumnNames(catalog, CallableHandles.SP_COLUMNS, arguments,
-                    getColumnsColumnNamesKatmai);
-        else
-            rs = getResultSetWithProvidedColumnNames(catalog, CallableHandles.SP_COLUMNS, arguments,
-                    getColumnsColumnNames);
-        // Hook in a filter on the DATA_TYPE column of the result set we're
-        // going to return that converts the ODBC values from sp_columns
-        // into JDBC values.
-        rs.getColumn(5).setFilter(new DataTypeFilter());
+            return rs;
+        } else {
```

Comment:
Need to add `switchCatalogs(catalog);` call here to make sure the behavior is consistent with Azure DB.

Reply:
Or remove `switchCatalogs(catalog);` for Azure DB too.